tests/phpunit/tests/rest-api/rest-schema-validation.php

```diff
@@ -348,4 +348,102 @@ public function test_object_or_string() {
 		$this->assertTrue( rest_validate_value_from_schema( array( 'raw' => 'My Value' ), $schema ) );
 		$this->assertWPError( rest_validate_value_from_schema( array( 'raw' => array( 'a list' ) ), $schema ) );
 	}
+
+	/*
+	 * @ticket 48818
+	 */
+	public function test_property_is_required() {
+		$schema = array(
+			'type'       => 'object',
+			'properties' => array(
+				'my_prop'          => array(
+					'type' => 'string',
+				),
+				'my_required_prop' => array(
+					'type'     => 'string',
+					'required' => 'true',
+				),
+			),
+		);
+		$this->assertWPError( rest_validate_value_from_schema( json_decode( '{"my_prop":"test"}' ), $schema ) );
+		$this->assertTrue( rest_validate_value_from_schema( json_decode( '{"my_required_prop":"test", "my_prop":"test"}' ), $schema ) );
+		$this->assertTrue( rest_validate_value_from_schema( json_decode( '{}' ), $schema ) );
+		$this->assertTrue( rest_validate_value_from_schema( json_decode( '[]' ), $schema ) );
+	}
+
+	/*
+	 * @ticket 48818
+	 */
+	public function test_property_is_required_v4() {
+		$schema = array(
+			'type'       => 'object',
+			'properties' => array(
+				'my_prop'          => array(
+					'type' => 'string',
+				),
+				'my_required_prop' => array(
+					'type' => 'string',
+				),
+			),
+			'required'   => array( 'my_required_prop' ),
+		);
+		$this->assertWPError( rest_validate_value_from_schema( json_decode( '{"my_prop":"test"}' ), $schema ) );
+		$this->assertTrue( rest_validate_value_from_schema( json_decode( '{"my_required_prop":"test", "my_prop":"test"}' ), $schema ) );
+		$this->assertTrue( rest_validate_value_from_schema( json_decode( '{}' ), $schema ) );
+		$this->assertTrue( rest_validate_value_from_schema( json_decode( '[]' ), $schema ) );
+	}
+
+	/*
+	 * @ticket 48818
+	 */
+	public function test_nested_property_is_required() {
+		$schema = array(
+			'type'       => 'object',
+			'properties' => array(
+				'my_object' => array(
+					'type'       => 'object',
+					'properties' => array(
+						'my_nested_prop'          => array(
+							'type' => 'string',
+						),
+						'my_required_nested_prop' => array(
+							'type'     => 'string',
+							'required' => 'true',
```

Comment:
```suggestion
							'required' => true,
```

A literal boolean here as well.

Reply:
done